OpenRA.Mods.Common/Traits/World/ShroudRenderer.cs

```diff
@@ -205,12 +209,12 @@ void IWorldLoaded.WorldLoaded(World w, WorldRenderer wr)
 			fogLayer = new TerrainSpriteLayer(w, wr, fogSheet, fogBlend, wr.Palette(info.FogPalette), false);
 		}
 
-		Edges GetEdges(PPos puv, Func<PPos, bool> isVisible)
+		Edges GetEdges(MPos uv, Func<PPos, bool> isVisible)
```

Comment:
Can we please keep this as `PPos` and do the casts where `RenderShroud` calls this? I would then also appreciate it if we could have a `// PERF: PPos coordinates are treated as MPos for visibility lookups` comment somewhere around the definition or use of `cellsDirty`.

We had a lot of trouble in the past with bugs caused by calculations done in the wrong coordinate systems, and it took a lot of effort plus these types to solve them.

Reply:
Alright.